src/java/org/apache/cassandra/net/SensorsCustomParams.java

```diff
@@ -154,20 +186,49 @@ public static <T> void addReadSensorToResponse(Message.Builder<T> response, Requ
+     */
+    public static <T> void addInternodeBytesSensorToResponse(Message.Builder<T> response, RequestSensors sensors, Context context)
+    {
+        addSensorToResponse(response, sensors, context, Type.INTERNODE_BYTES,
+                            (sensor) -> SensorsCustomParams.encodeTableInInternodeBytesRequestParam(sensor.getContext().getTable()),
+                            (sensor) -> SensorsCustomParams.encodeTableInInternodeBytesTableParam(sensor.getContext().getTable()));
+    }
+
+    /**
+     * A utility method to encode internode bytes sensor values in the internode response message. This method will iterate
+     * over all the write sensors in the {@link RequestSensors} and encode the sensor values in the internode response message.
+     */
+    public static <T> void addInternodeBytesSensorToResponse(Message.Builder<T> response, RequestSensors sensors)
+    {
+        for (Sensor sensor : sensors.getSensors(Type.INTERNODE_BYTES))
+        {
+            addSensorToResponse(response, sensor,
+                                (s) -> SensorsCustomParams.encodeTableInInternodeBytesRequestParam(sensor.getContext().getTable()),
+                                (s) -> SensorsCustomParams.encodeTableInInternodeBytesTableParam(sensor.getContext().getTable()));
+        }
+    }
+
     private static <T> void addSensorToResponse(Message.Builder<T> response, RequestSensors sensors, Context context, Type type,
                                                 Function<Sensor, String> requestParamSupplier,
                                                 Function<Sensor, String> tableParamSupplier)
     {
         Optional<Sensor> requestSensor = sensors.getSensor(context, type);
-        requestSensor.ifPresent(sensor -> {
-            byte[] requestBytes = SensorsCustomParams.sensorValueAsBytes(sensor.getValue());
-            response.withCustomParam(requestParamSupplier.apply(sensor), requestBytes);
-
-            Optional<Sensor> registrySensor = SensorsRegistry.instance.getSensor(sensor.getContext(), type);
-            registrySensor.ifPresent(registry -> {
-                byte[] tableBytes = SensorsCustomParams.sensorValueAsBytes(registry.getValue());
-                response.withCustomParam(tableParamSupplier.apply(registry), tableBytes);
-            });
+        requestSensor.ifPresent(sensor -> addSensorToResponse(response, sensor, requestParamSupplier, tableParamSupplier));
+    }
+
+    private static <T> void addSensorToResponse(Message.Builder<T> response, Sensor sensor,
+                                                Function<Sensor, String> requestParamSupplier,
+                                                Function<Sensor, String> tableParamSupplier)
+    {
+        byte[] requestBytes = SensorsCustomParams.sensorValueAsBytes(sensor.getValue());
+        response.withCustomParam(requestParamSupplier.apply(sensor), requestBytes);
```

Comment:
I am confused why is `requestBytes` value obtained from `Sensor` stored in `RequestSensors` and few lines below `tableBytes` it's obtained from Sensor stored in the Registry?

Reply:
`requestBytes` are per request & per table. `tableBytes` are monotonically increasing summing up bytes from all requests, per table which is what the sensor registry is for.

Reply:
Yes I get that SensorRegistry holds aggregate value across all requests since start, but here we are attaching sensor bytes to the response. Why would a single response have a cumulative value?

Reply:
The idea was (when we started UCU project) it to rollup UCUs at the coordinator (or cql router) level across and make decision like rate limiting. For that we need to calculate "rate of change" which makes sense across requests. One can argue that coordinator itself could keep a request of the "per request UCU" to deduce the registry one, but if we spin up a new coordinator, having the writer return the rate (or sufficient info to calculate the rate which is the sensor registry value) will enable the coordinate to rate limit earlier. (See https://github.com/riptano/cndb/issues/8501, it also has a reference implementation with rate calculation, but then we put the interaction with coordinator on hold until CQL router completes) 

It is worth mentioning that all UCU we have today follow this convention (send per request and across request value).

Reply:
Thanks for clarifying. I am still a bit confused with that design decision but lets put that aside for this PR (we can discuss that more offline).